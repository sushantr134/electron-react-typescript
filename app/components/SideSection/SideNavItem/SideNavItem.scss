@import '../../../app.global';

a.selectedLink {
  color: $lt-font-color !important;
}
