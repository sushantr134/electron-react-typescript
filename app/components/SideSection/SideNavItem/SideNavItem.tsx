import * as React from 'react';
import { Link } from 'react-router-dom';
import styles from './SideNavItem.scss';
import variables from '../../../app.global.scss';
import Pipeline from '../../common/Icons/Pipeline';
import Terminal from '../../common/Icons/Terminal';
import HexagonIcon from '../../common/Icons/Hexagon';

export type MenuItem = {
  text: string;
  id: number;
  icon: any;
  to: string;
};

type Props = {
  menu: MenuItem;
  isActive: boolean;
  onMenuClick: (e: React.MouseEvent<HTMLLIElement>) => void;
};
export class SideNavItem extends React.PureComponent<Props> {
  props: Props;
  size = 26;

  anchorClick = (e: React.FormEvent<HTMLAnchorElement>) => {
    e.preventDefault();
  };

  getIcon = (id: number) => {
    let navIcon;
    switch (id) {
      case 1:
        if (this.props.isActive) {
          navIcon = (
            <HexagonIcon color={variables.ltfontcolor} size={this.size} />
          );
        } else {
          navIcon = (
            <HexagonIcon color={variables.ltfontdull} size={this.size} />
          );
        }
        break;
      case 2:
        if (this.props.isActive) {
          navIcon = <Pipeline color={variables.ltfontcolor} size={this.size} />;
        } else {
          navIcon = <Pipeline color={variables.ltfontdull} size={this.size} />;
        }
        break;
      case 3:
        if (this.props.isActive) {
          navIcon = <Terminal color={variables.ltfontcolor} size={this.size} />;
        } else {
          navIcon = <Terminal color={variables.ltfontdull} size={this.size} />;
        }
        break;
    }
    return navIcon;
  };

  render() {
    const { text, id, to } = this.props.menu;
    const navIcon = this.getIcon(id);
    let clsName;
    if (this.props.isActive) {
      clsName = styles.selectedLink;
    }
    return (
      <li data-id={id} onClick={this.props.onMenuClick}>
        <Link to={to} className={clsName} onClick={this.anchorClick}>
          {navIcon}

          <span>{text}</span>
        </Link>
      </li>
    );
  }
}
