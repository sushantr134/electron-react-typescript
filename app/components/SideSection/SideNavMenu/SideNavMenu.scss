@import '../../../app.global';

.sideMenu {
  margin: 0;
  list-style: none;
  padding-left: 0;
}

.sideMenu > li {
  padding: 8px 15px;
  cursor: pointer;
  transition: all 0.5s;
}

li.selected {
  background-color: $lt-highlighted-tab;
}

.sideMenu > li:hover {
  background-color: $lt-highlighted-tab;
}

.sideMenu > li > a {
  display: table;
  text-decoration: none;
  color: $lt-font-dull-color;
}

.sideMenu > li > a > span {
  padding-left: 8px;
  display: table-cell;
  vertical-align: middle;
}

.sideMenu > li > a > img {
  height: 32px;
  width: 32px;
}
