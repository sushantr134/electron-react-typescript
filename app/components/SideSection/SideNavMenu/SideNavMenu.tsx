import * as React from 'react';
import { PureComponent } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { MenuItem } from '../SideNavItem/SideNavItem';
import { SideNavItem } from '../SideNavItem/SideNavItem';
import styles from './SideNavMenu.scss';

interface Props extends RouteComponentProps<any> {
  menuItems: Array<MenuItem>;
  defaultSelected: number;
  history: any;
}

type State = {
  activeMenuId: number;
};

class SideNavMenu extends PureComponent<Props, State> {
  props: Props;
  state: State;

  onNavMenuItemClick = (e: React.MouseEvent<HTMLLIElement>) => {
    e.preventDefault();
    const id = e.currentTarget.dataset.id;
    e.stopPropagation();
    const selectedMenu: Array<MenuItem> = this.props.menuItems.filter(
      elem => elem.id == Number(id)
    );
    this.setState({
      activeMenuId: Number(id)
    });
    this.props.history.push(selectedMenu[0].to);
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      activeMenuId: 1
    };
  }

  render() {
    const menuItems = this.props.menuItems.map(item => (
      <SideNavItem
        key={item.id}
        menu={item}
        isActive={item.id == this.state.activeMenuId}
        onMenuClick={this.onNavMenuItemClick}
      />
    ));
    return <ul className={styles.sideMenu}>{menuItems}</ul>;
  }
}

export default withRouter(SideNavMenu);
