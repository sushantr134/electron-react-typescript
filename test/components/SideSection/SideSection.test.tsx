import * as React from 'react';
import { mount, shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import SideNavMenu from '../../../app/components/SideSection/SideNavMenu/SideNavMenu';
import { MenuItem } from '../../../app/components/SideSection/SideNavItem/SideNavItem'; //importing types for menu items
import { MemoryRouter } from 'react-router';

describe('Testing of SideSection Component', () => {
  let SideNavMenuTestComponent: any;

  const menuItems: Array<MenuItem> = [
    {
      text: 'Nav Item 1',
      id: 1,
      icon: '',
      to: ''
    },
    {
      text: 'Nav Item 2',
      id: 2,
      icon: '',
      to: '/launchpad'
    }
  ];

  beforeEach(() => {
    SideNavMenuTestComponent = (
      <MemoryRouter>
        <SideNavMenu menuItems={menuItems} defaultSelected={1} />
      </MemoryRouter>
    );
  });

  it('rendering SideNavItems correctly', () => {
    const component = renderer.create(SideNavMenuTestComponent).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('checking first navItemName', () => {
    const wrapper = mount(SideNavMenuTestComponent);
    expect(
      wrapper
        .find('a')
        .first()
        .text()
    ).toBe('Nav Item 1');
  });

  it('Checking no. of List Items rendered based on props data', () => {
    const wrapper = mount(SideNavMenuTestComponent);
    expect(wrapper.find('.sideMenu').find('li').length).toBe(1); //test must be failed because we have 2 items in props above defined in var menuItems
  });
});
