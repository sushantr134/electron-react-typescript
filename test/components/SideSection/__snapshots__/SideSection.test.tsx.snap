// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of SideSection Component rendering SideNavItems correctly 1`] = `
<ul
  className="sideMenu"
>
  <li
    data-id={1}
    onClick={[Function]}
  >
    <a
      className="selectedLink"
      href="/"
      onClick={[Function]}
    >
      <svg
        height={26}
        id="Capa_1"
        viewBox="0 0 490 490"
        width={26}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="ltfontcolor"
        >
          <path
            d="M466.032,367.5v-245L245,0L23.968,122.5v245L245,490L466.032,367.5z M118.865,175.094L245,105.188l126.136,69.906v139.813 L245,384.813l-126.135-69.906V175.094z"
          />
          <polygon
            points="331.299,292.828 331.299,197.172 245,149.343 158.701,197.172 158.701,292.828 245,340.657"
          />
        </g>
      </svg>
      <span>
        Nav Item 1
      </span>
    </a>
  </li>
  <li
    data-id={2}
    onClick={[Function]}
  >
    <a
      href="/launchpad"
      onClick={[Function]}
    >
      <svg
        height={26}
        id="Capa_1"
        viewBox="0 0 52.371 52.37"
        width={26}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="ltfontdull"
        >
          <path
            d="M26.185,0c0,0-10.073,7.816-10.073,27.016c0,12.314,3.724,20.015,8.573,21.33V35.951c0-0.828,0.671-1.5,1.5-1.5 c0.829,0,1.5,0.672,1.5,1.5v12.395c4.849-1.315,8.572-9.016,8.572-21.33C36.257,7.594,26.185,0,26.185,0z M26.066,14.818 c-1.729,0-3.418-0.155-4.914-0.438c1.465-4.721,3.531-7.901,5.051-9.761c1.516,1.837,3.571,4.989,5.026,9.715 C29.677,14.648,27.896,14.818,26.066,14.818z"
          />
          <path
            d="M13.408,35.504c-3.007,0-5.444,2.081-5.444,4.648V52.37l9.862-7.296c-1.463-2.496-2.556-5.737-3.201-9.57H13.408z"
          />
          <path
            d="M38.96,35.504h-1.216c-0.646,3.833-1.738,7.074-3.201,9.57l9.862,7.296V40.152C44.405,37.585,41.968,35.504,38.96,35.504 z"
          />
        </g>
      </svg>
      <span>
        Nav Item 2
      </span>
    </a>
  </li>
</ul>
`;
